### DISCUSSION TITLE
Manage multi-steps form (with validity check flow) and watch errors without re-rendering whole form

### DISCUSSION THREAD

**Discussion by moulinraphael** [react-hook-form/react-hook-form#1979] [Ideas]
Hey everyone ! 

For a personal project, I needed to :
- **handle multi-steps form** (with some check on the validity of step's fields)
- **view errors live** (without re-render because in the current API, `errors` are inside `formState`)

So, I implemented a draft to manage these points. It's not perfect and I do not have a lot of time to spent but it could be interesting for some of you or maybe for some future features of this great lib (thanks the team and @bluebill1049 )

https://codesandbox.io/s/react-hook-form-form-extended-cteuy

Let me know if you have any comment, suggestion :)

![image](https://user-images.githubusercontent.com/4943583/85932991-24e74280-b8d2-11ea-8c62-c4cbdb1640ea.png)

**Comment by bluebill1049** [comment]
Thanks @moulinraphael Nice one, I think there is a user was asking for similar usage. I will share this.
